### PR TITLE
feat: per-library TV metadata source (TVDB/TMDB) with provider-aware refresh

### LINKED ISSUE #180: [FR] Allow forcing TMDB as the metadata provider for TV libraries as option
Currently TV shows are hardcoded to use TVDB for metadata.

TVDB has some long-standing issues as a TV source:

- It frequently re-orders absolute/episode numbering. Specials get inserted into or removed from the order, which shifts every subsequent episode and breaks matching — this happens especially often with anime, sometimes every few weeks.
- It tracks aired order by default, which often doesn't match DVD/production order, so episodes end up mismatched.
- Specials (Season 00) numbering and ordering is inconsistent.

I'd like an option to use TMDB for TV shows instead, so metadata comes from a single, more stable source.

### PR DESCRIPTION
**Pull request by arsfeld**
## Summary

Adds a per-library **TV metadata source** setting (TVDB or TMDB, default TVDB), addressing #180. New shows match against their library's configured provider, and the existing refresh-metadata action becomes provider-aware: when a show's stored data came from a different provider than its library now uses, refresh re-identifies the show against the library's provider, falling back to a manual picker when it can't confidently auto-match.

This keeps TVDB as the default for everyone (no migration churn) while letting operators put, say, an anime library on TMDB when TVDB's ordering churn breaks matching.

## Key decisions

- **Per-library granularity.** The provider is a property of a TV library, not global or per-show — fitting a "separate anime library" setup. Default stays TVDB; nothing changes until an operator opts a library in.
- **Authoritative `metadata_source` column.** A TV show can legitimately carry both `tvdb_id` and `tmdb_id`, so provenance is recorded explicitly (backfilled from `metadata.provider_id`, not inferred from id presence) and is the single source of truth for the refresh decision.
- **Safe, transactional episode reconciliation.** Switching a show's provider fetches and validates all new-provider season data *before* deleting anything, then wipes + recreates episodes and re-links files by filename inside one transaction. A failed/empty/partial fetch aborts with the old episodes intact; files are re-attached to the show rather than orphaned.
- **Conservative auto-adopt gate.** A silent switch requires a near-exact title **and** a real year match on both sides; anything looser (including a missing year) routes to the manual picker.
- **Episode-level watch state resets on a switch** (`ON DELETE SET NULL`); the operator is warned in the confirmation copy / flash.

## What changed

- **Library setting + UI** — `tv_metadata_source` column on `library_paths` (default TVDB) and a provider dropdown gated to series/mixed libraries.
- **Provenance** — `metadata_source` column on `media_items` with a provider-aware backfill; `maybe_discover_tvdb_id` no longer disturbs it.
- **New-match routing** — `Relay.search/3` is provider-aware; the scan and import paths thread the library's provider into matching and stamp `metadata_source`.
- **Provider-aware refresh** — `Mydia.Media.ProviderSwitch` (decision, candidate finder, transactional `adopt_provider_switch/4`), wired into the show LiveView with a manual re-identify picker. Search + switch run via `start_async` so the socket never blocks.

## Testing

- Unit/integration tests for: the library + media_item schema/changesets, the backfill, provider-aware relay search, `metadata_source` stamping (Bypass), the refresh decision + library resolution (incl. episode-linked files + ambiguous), the candidate confidence gate, and the transactional reconciliation — including the failure/partial-fetch/id-collision/yearless edge cases that preserve data.
- Full suite green (1463 tests, 0 failures). Also fixes a pre-existing test-isolation bug where `registry_test` left the provider registry cleared for later tests.

## Known residuals (follow-ups, not regressions)

- Season/search/fetch cache keys omit the provider; a numeric id shared across TVDB/TMDB could collide (pre-existing caching behavior, low probability — ids rarely collide across providers).
- Added coverage is focused on the data-mutation paths; some completeness gaps remain (e.g. the TVDB-target adopt path, the year-retry search branch, the enricher `update_existing` path) and the full LiveView refresh→picker flow is covered at the component/context level rather than end-to-end.

## Post-Deploy Monitoring & Validation

- **Migrations:** two additive migrations (`library_paths.tv_metadata_source`, `media_items.metadata_source` + backfill). Both reversible; run on SQLite and Postgres. After deploy, confirm no TV shows ended up with zero episodes after a switch:
  ```sql
  SELECT mi.id, mi.title, mi.metadata_source, COUNT(e.id) AS episode_count
  FROM media_items mi LEFT JOIN episodes e ON e.media_item_id = mi.id
  WHERE mi.type = 'tv_show' GROUP BY mi.id, mi.title, mi.metadata_source
  HAVING COUNT(e.id) = 0;
  ```
- **Watch / log:** a provider switch logs `Provider switched to <provider>`. Watch for `:provider_switch_update_failed` (id collision) and `:season_fetch_failed` / `:missing_tvdb_season_id` (aborted switches, old data preserved).
- **Healthy signal:** flipping a library to TMDB changes nothing until a show is refreshed; refreshing a mismatched show either auto-switches (confident) or opens the picker. **Failure signal:** a show left episode-less after a refresh (should be impossible given the abort-on-partial-fetch guard) → investigate the relay's season responses.

Closes #180

## Update: discover-add provenance & source selection

Closes the discover/add gap left by the original change. Adding a TV show from `/discover` fully enriches it (content + episodes) but previously hardcoded TVDB and never stamped `metadata_source`, so shows added against a TMDB library came in TVDB-sourced and unstamped.

- **Derive on add** — `Settings.derive_tv_metadata_source/0` returns the unanimous source across enabled series/mixed libraries (`:tvdb` when none, `nil` on conflict). Discover-add fetches from the derived provider and stamps `metadata_source` so content, episodes, and provenance agree; conflicts leave it `nil` for the scan path to resolve. Movies stay nil. The detail-modal preview reflects the same source.
- **Scan-path self-heal** — the enricher previously dropped the match's provider before the update path, so nil-source items fell back to id-inference and the library setting never got a vote. It now adopts the match provider for nil-source items (including within the recently-enriched window, stamp-only), preserves an already-recorded source, and drops the dead id-inference fallback. Pre-existing nil-source items self-heal to the library's source on next scan; no migration.

This resolves the prior residual noting the enricher `update_existing` path and discover-add were uncovered. Tests added for the derive helper, discover-add stamping (Bypass), and the enricher nil-source adoption incl. the freshness-window and preserve-behavior regression guard.

### CODE REVIEW

File: lib/mydia_web/live/media_live/show/file_events.ex [hunk 1]
```diff
@@ -13,13 +14,138 @@ defmodule MydiaWeb.MediaLive.Show.FileEvents do
...
+         socket
+         |> assign(:reidentifying, true)
+         |> put_flash(:info, "Re-identifying on #{provider_label(target)}...")
+         |> start_async(:reidentify_search, fn ->
+           {target, ProviderSwitch.find_reidentify_candidate(media_item, target)}
+         end)}
+
+      :refetch ->
+        do_standard_refresh(media_item, socket)
+    end
```
[the remark this answers is not in the document]
Comment (arsfeld) [reply lib/mydia_web/live/media_live/show/file_events.ex:38]: Good catch, this was a real correctness gap. Fixed in 1269504d: the standard refresh now resolves its provider from the authoritative `metadata_source` column first (via `refresh_provider_preference/1`), so a TMDB-sourced show with a back-filled `tvdb_id` refreshes from TMDB. The legacy tvdb-precedence rule only applies when `metadata_source` is nil, and TVDB-id discovery in `refresh_episodes_for_tv_show/2` is now skipped whenever a show has an explicit source. Added a regression test in `media_test.exs` ("refresh_metadata/1 provider routing").

File: lib/mydia/library/metadata_enricher.ex [hunk 2]
```diff
@@ -218,18 +226,35 @@ defmodule Mydia.Library.MetadataEnricher do
...
+    # TMDB-matched show is not fetched from the TVDB endpoint (and vice versa).
+    fetch_opts =
+      if media_type == :tv_show && provider_type in [:tvdb, :tmdb] do
+        Keyword.put(fetch_opts, :provider, provider_type)
+      else
+        fetch_opts
+      end
+
     Metadata.fetch_by_id_cached(config, provider_id, fetch_opts)
   end
```
[the remark this answers is not in the document]
Comment (arsfeld) [reply lib/mydia/library/metadata_enricher.ex:275]: Fixed in 1269504d by adding the provider to the `fetch_by_id_cached` cache key (`fetch_by_id:#{provider}:#{provider_id}:...`), defaulting to the config type when no `:provider` opt is given. Overlapping TVDB/TMDB ids of the same media_type can no longer collide in the cache.

File: priv/repo/migrations/20260604120100_add_metadata_source_to_media_items.exs [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+        nil ->
+          :ok
+
+        source ->
+          repo().update_all(
+            from(m in "media_items", where: m.id == ^row.id),
+            set: [metadata_source: source]
+          )
+      end
+    end)
```
[the remark this answers is not in the document]
Comment (arsfeld) [reply priv/repo/migrations/20260604120100_add_metadata_source_to_media_items.exs]: Leaving the backfill as-is. Mydia is self-hosted at personal-library scale (hundreds to a few thousand shows), the migration runs once, and each row selects only `id`, the two id columns, and `metadata`. That stays well within memory and the migration timeout for the target deployment, so streaming would add transaction-scoping complexity (SQLite + Postgres) without a real payoff here.
